app/src/main/java/com/allan/boardbuddies/fragments/BoardsFragment.java

```diff
@@ -90,36 +74,29 @@ protected void populateMemoViewHolder(MemoViewHolder holder, Board element) {
     public void onViewCreated (View view, Bundle savedInstanceState){
         FloatingActionButton addNoteFab = view.findViewById(R.id.add_note_fab);
         addNoteFab.setOnClickListener(v -> {
-            Intent intent = new Intent(v.getContext(), BoardActivity.class);
-            intent.putExtra("FILEPATH", directory);
-            resultLauncher.launch(intent);
+            onMemoClick(-1);
+        });
+        memoListViewModel.getBoards().observe(getViewLifecycleOwner(), boards ->{
+            adapter.submitList(boards);
         });
     }
 
-    public void onElementClick(int position){
-        Intent intent = new Intent(requireContext(), BoardActivity.class);
-        intent.putExtra("FILENAME", boards.get(position).getFileName());
-        intent.putExtra("FILEPATH", directory);
-        intent.putExtra("POSITION", position);
-        resultLauncher.launch(intent);
+    public void onMemoClick(int position){
+        Bundle bundle = new Bundle();
+        bundle.putInt("position", position);
+        Navigation.findNavController(requireActivity(), R.id.nav_host).navigate(R.id.action_to_editBoardFragment, bundle);
     }
 
-    private void loadSingle(String filename, int position){
-        String fileString = Utilities.getFileAsString(new File(directory, filename));
-        boards.add(position, new Gson().fromJson(fileString, Board.class));
-    }
+    private static class BoardDiffCallback<Board> extends DiffUtil.ItemCallback<Board> {
+        @Override
+        public boolean areItemsTheSame(Board oldItem, Board newItem) {
+            return oldItem.equals(newItem);
+        }
 
-    private void loadNotes(){
-        File[] files = directory.listFiles();
-        boards.clear();
-        if (files != null) {
-            Arrays.sort(files, Comparator.comparing(File::getName).reversed());
-            for (File file : files) {
-                if (file.isFile() && file.getName().endsWith(".json")) {
-                    String fileString = Utilities.getFileAsString(file);
-                    boards.add(new Gson().fromJson(fileString, Board.class));
-                }
-            }
+        @Override
+        public boolean areContentsTheSame(Board oldItem, Board newItem) {
```

Comment:
what's the difference between this and `areItemsTheSame` above?

Reply:
areItemsTheSame - used to determine structural changes between old and new list (additions/removals/position changes)

areContentsTheSame - determines if particular item was updated